ui/src/components/popovers/OCRModal.tsx

```diff
@@ -0,0 +1,168 @@
+import { useCallback, useMemo, useState } from "react";
+import { LuCornerDownLeft } from "react-icons/lu";
+import { ExclamationCircleIcon } from "@heroicons/react/16/solid";
+import { type LoggerMessage } from "tesseract.js";
+
+import { Button } from "@components/Button";
+import { GridCard } from "@components/Card";
+import { TextAreaWithLabel } from "@components/TextArea";
+import { SettingsPageHeader } from "@components/SettingsPageheader";
+import { useUiStore } from "@/hooks/stores";
+import useOCR from "@/hooks/useOCR";
+import { cx } from "@/cva.config";
+
+export default function OCRModal({ videoElmRef }: { videoElmRef?: React.RefObject<HTMLVideoElement | null> }) {
+  const { setDisableVideoFocusTrap } = useUiStore();
+  const [ocrStatus, setOcrStatus] = useState<LoggerMessage>();
+  const [ocrError, setOcrError] = useState<string | null>(null);
+  const handleOcrError = useCallback((error: any) => { // eslint-disable-line @typescript-eslint/no-explicit-any
+    if (typeof error === "string") {
+      setOcrError(error);
+    } else {
+      setOcrError(error.message);
+    }
+  }, [setOcrError]);
+  const [ocrText, setOcrText] = useState<string | null>(null);
+
+  const { ocrImage } = useOCR();
+
+  const onConfirmOCR = useCallback(async () => {
+    setDisableVideoFocusTrap(true);
+
+    setOcrText(null);
+    setOcrError(null);
+    setOcrStatus(undefined);
+
+    if (!videoElmRef?.current) {
+      setOcrError("Video element not found");
+      return;
+    }
+
+    setOcrStatus({
+      status: "Capturing image",
+      progress: 0,
+      jobId: "",
+      userJobId: "",
+      workerId: "",
+    });
+
+    // create a canvas from the video element then capture the image from the canvas
+    const video = videoElmRef?.current;
+    const canvas = document.createElement("canvas");
+    canvas.width = video.videoWidth;
+    canvas.height = video.videoHeight;
+
+    const ctx = canvas.getContext("2d");
+    ctx?.drawImage(video, 0, 0, canvas.width, canvas.height);
```

Comment:
Would we want to "pause" the video before and "unpause" the video after this `drawImage` call to ensure we get a full frame instead of a (possibly partial) live view?